src/Events/UserInterface/PageHeadingRenderEvent.php

```diff
@@ -98,7 +99,22 @@ public function setPrimaryMenuItem(MenuItemInterface $item, ?int $position = nul
 
     public function getPrimaryMenuItems(): MenuItems
     {
-        return $this->primary_menu;
+        $items = $this->primary_menu;
+
+        foreach ($items as $k => $i) {
+            // No check if ACL element is absent
+            if (count($i->getAcl()) == 0) {
+                continue;
+            }
+
+            // Unset the item if the user does not have access
+            $acl = $i->getAcl();
+            $rv = (count($acl) > 2) ? $acl[2] : [];
```

Comment:
if acl[2] is not set, then would set it to empty string instead of an empty array